WordPress/Classes/ViewRelated/Reader/ReaderMenuViewController.swift

```diff
@@ -0,0 +1,172 @@
+    ///
+    /// - Parameters:
+    ///     - topic: The topic to show.
+    ///
+    func showPostsForTopic(topic: ReaderAbstractTopic) {
+        let controller = ReaderStreamViewController.controllerWithTopic(topic)
+        navigationController?.pushViewController(controller, animated: true)
+    }
+
+
+    /// Presents the reader's search view controller.
+    ///
+    func showReaderSearch() {
+        let controller = ReaderSearchViewController.controller()
+        navigationController?.pushViewController(controller, animated: true)
+    }
+
+
+    // MARK: - TableView Delegate Methods
+
+
+    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return viewModel.numberOfSectionsInMenu()
+    }
+
+
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.numberOfItemsInSection(section)
+    }
+
+
+    override func tableView(tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let title = viewModel.titleForSection(section)
+        let header = WPTableViewSectionHeaderFooterView(reuseIdentifier: nil, style: .Header)
+        header.title = title
+        return header
+    }
+
+
+    override func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        let title = viewModel.titleForSection(section)
+        return WPTableViewSectionHeaderFooterView.heightForHeader(title, width: view.frame.width)
+    }
+
+
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier(cellIdentifier)!
+        configureCell(cell, atIndexPath: indexPath)
+        return cell
+    }
+
+
+    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        guard let menuItem = viewModel.menuItemAtIndexPath(indexPath) else {
+            return
+        }
+
+        // TODO: Remember selection
+
+        if let topic = menuItem.topic {
+            showPostsForTopic(topic)
+            return
+        }
+
+        if menuItem.type == .Search {
+            showReaderSearch()
+            return
+        }
+    }
+
+
+    func configureCell(cell: UITableViewCell, atIndexPath indexPath: NSIndexPath) {
+        guard let menuItem = viewModel.menuItemAtIndexPath(indexPath) else {
+            return
+        }
+        WPStyleGuide.configureTableViewCell(cell)
+        cell.accessoryType = .DisclosureIndicator
+        cell.selectionStyle = .Default
+        cell.textLabel?.text = menuItem.title
+        cell.imageView?.image = menuItem.icon
+    }
+
+}
+
+
+extension ReaderMenuViewController : ReaderMenuViewModelDelegate
+{
+
+    func menuWillReloadContent() {
+
+    }
+
+
+    func menuDidReloadContent() {
+        tableView.reloadData()
+    }
+
+
+    func menuSectionWillChangeContent(index: Int) {
```

Comment:
Actually, I think I'll remove them outright. If it turns out the methods are needed we can bring them back.